proto/viam/app/mltraining/v1/ml_training.proto

```diff
@@ -39,7 +38,6 @@ message GetTrainingJobRequest {
 }
 
 message GetTrainingJobResponse {
-  string id = 1;
   TrainingJobMetadata metadata = 2;
```

Comment:
change this number to 1

Reply:
This can actually cause issues if for example there are robots or servers on app running older versions of rdk, since they will expect gRPC responses where metadata corresponds to index 2, but app will send a message with metadata as field 1. I don't know how or how often rdk updates on robots but I assume they don't all update simultaneously. So we might want to change this at a later point or design this for backwards compatibility. How we handle this depends on which of robots/users/app frontend actually send these requests and whether or not we can be in a state where the backend is running the latest version of app but clients are sending gRPC requests with older versions of rdk or app frontend.

Reply:
In general I'd totally agree with the motivation to maintain backwards compatibility, but in this case I think:

- The only thing using this is app, and we control the version + deployment there.
- No other clients are using this, so there isn't a risk of a version mismatch.

In this case I think the risk of the breaking change is less than the benefit of keeping the proto as clean as possible for as long as possible